packages/flutter/lib/src/material/stepper.dart

```diff
@@ -140,6 +140,7 @@ class Stepper extends StatefulWidget {
     this.onStepTapped,
     this.onStepContinue,
     this.onStepCancel,
+    this.controlsBuilder,
```

Comment:
I did not realize users had to pass in the step callbacks and current state already, which means there is no need to pass them through the callback - I spaced on this a bit, really sorry!

Instead the API should look something like this:

```dart
// Assume currentStep, onTapped, onContinue, and onCancel are defined as members of a State object.
  return Stepper(
    steps: [ ... ],
    type: StepperType.vertical,
    currentStep: currentStep,
    onStepTapped: onTapped ,
    onStepContinue: onContinue,
    onStepCancel: onCancel,
    controls: (BuildContext context) {
      return Row(children: [Button(onPressed: onContinue, ...), Button(onPressed: onCancel, ...)]);
    },
  );
```

Reply:
I don't think so? Those parameters are optional so we need to provide a way for the user to have access to the default functionality if they don't pass in those functions. I guess we could change the stepper to require those functions to be set, but I feel like the implementation that we have right now works well. I'm also realizing the current implementation should also function correctly in the case that these functions are set to a value or to null.

Reply:
So just to clarify this conversation, because this extended functionality is primarily about aesthetics, it makes the most sense to not let the user pass in their own functions and instead make their buttons use the two functions already included in the stepper.